docs/sql-functions/approximate-aggregate/approx-topk.md

```diff
@@ -0,0 +1,169 @@
+```json
+{
+  "item": [ { "value": "192.168.1.100", "count": 2650 }, { "value": "10.0.0.5", "count": 2230 }, { "value": "203.0.113.50", "count": 2210 }, { "value": "198.51.100.75", "count": 1979 }, { "value": "172.16.0.10", "count": 1939 } ]
+}
+```
+
+### Use `approx_topk` With `unnest`
+To convert these nested results into individual rows, use the `unnest()` function.
+
+```sql
+SELECT item.value as clientip, item.count as request_count 
+FROM ( 
+    SELECT unnest(approx_topk(clientip, 20)) as item 
+    FROM "demo1" 
+    ) 
+ORDER BY request_count 
+DESC
+```
+**Result of `approx_topk()` with `unnest()`**
+This provides a flat output as shown below: 
+
+```json
+{ "value": "192.168.1.100", "count": 2650 }
+{ "value": "10.0.0.5", "count": 2230 }
+{ "value": "203.0.113.50", "count": 2210 }
+...
+```
+
+---
+
+## `GROUP BY` versus `approx_topk`
+
+### How GROUP BY Works
+The traditional way to find the top values in a field is by using a `GROUP BY` query combined with `ORDER BY` and `LIMIT`. <br>
+    For example:
+
+    ```sql
+
+    SELECT clientip AS x_axis_1, COUNT(*) AS y_axis_1 
+    FROM cdn_production 
+    GROUP BY x_axis_1 
+    ORDER BY y_axis_1 DESC 
+    LIMIT 10
+    ```
+    This query counts how many times each unique `clientip` appears and returns the **top 10** based on that count.
+
+??? info "Why Traditional `GROUP BY` Breaks in Large Datasets"
+    In large datasets with high-cardinality fields, the query is executed across multiple querier nodes. Each node uses multiple CPU cores to process the data. The data is split into partitions, and each core handles a subset of partitions.
+
+    Consider the following scenario:
+
+    - Dataset contains `3 million` unique client IPs.
+    - Query runs using `60` CPU cores.
+    - Each core processes `60` partitions.
+
+    This results in:
+    
+    `3 million` values × `60` cores × `60` partitions = `10.8 billion` data entries being processed in memory.
+
+    This level of memory usage can overwhelm the system and cause failures.
+
+    **Typical Failure Message** <br>
+    ```
+    Resources exhausted: Failed to allocate additional 63232256 bytes for GroupedHashAggregateStream[20] with 0 bytes already allocated for this reservation - 51510301 bytes remain available for the total pool
+    ```
+    ![Typical Failure Message](../../images/approx-top-k-error-in-traditional-method.png)
+    This is a common limitation of using traditional `GROUP BY` with high-cardinality fields in large environments.
+
+### How `approx_topk` Works
+When you run a query using `approx_topk()`, each query node processes a subset of the dataset and computes its local approximate top K values. These local top K values are sent to the leader node. The leader node merges them to generate the final approximate result. 
+
+Because each node sends only its local top K values, the final result may miss values that are frequent across the entire dataset but do not appear in the top K list of any single node. 
```

Comment:
Each node not only send topK to leader, actually it will send
`max(k*10, 1000)`

it means, when you want to top 10, each node will send 1000 to the leader. 

why not only top10?

because maybe some item is not top 10 but in other node is top 10, we need some capacity to fix the final value.